### ISSUE TITLE
Use Electron `setContentProtection` API on Windows and macOS

### ISSUE THREAD

**Issue by link2xt** [deltachat/deltachat-desktop#4347]
Electron has a function to prevent screenshotting the application window:
https://www.electronjs.org/docs/latest/api/browser-window#winsetcontentprotectionenable-macos-windows

On Windows and macOS this could be enabled by default and and option to disable it provided.

Delta Chat on Android has a similar "Screen security" option in Advanced settings, so it makes sense to have it on desktop.

Windows is introducing "Recall" and if screen capture is not prevented, disappearing messages can easily end up in the history stored on device, affecting both the user and all its contacts: 
https://support.delta.chat/t/opt-out-of-windows-recall-screenshotting-the-chats-on-desktop/3342

**Comment by r10s** [comment]
might be a nice option, however, not sure about preventing screenshots it _by default._ some counter args:

screenshots are very much a thing todays, and it is questionable to make it harder to get them for the majority of users that want that feature. also on android, it is possible to do screenshots by default.

for windows "recall" - afaik, ppl has to agree for that feature before - and if they did, and are at risk, they have probably much larger problems

**Comment by r10s** [comment]
> The first thing is to have an option at all, even if disabled by default

agree, that is also a good way to get experience first, if there are side effect etc. eg. on android, that does not only avoid screenshots, but also worsens the "switch app" experience

**Comment by maxphilippov** [comment]
> Signal does not seems to call this function at all. Likely they are not aware it exists. Also Windows 11 is not widespread (yet), so there is some time before "AI-enabled" laptops are a threat.
I had Copilot installed (although I'm not sure if it's a full install or just a shortcut which will load a full version, I cleaned it up) as part of the latest Win10 security update, so there might be a chance they retroactively and silently enable that.

**Comment by Simon-Laux** [comment]
For reference, the api under tauri: https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.set_content_protected
